api/envoy/config/common/matcher/v3/matcher.proto

```diff
@@ -16,6 +19,80 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
 
 // [#protodoc-title: Unified Matcher API]
 
+message MatchTree {
+  message MatchAction {
+    oneof action {
+      bool skip = 1;
+
+      string callback = 2;
+    }
+  }
+
+  message MatchLeaf {
+    message LeafMatcher {
+      // The predicate to evaluate.
+      oneof matcher {
+        MatchPredicate predicate = 1;
+
+        core.v3.TypedExtensionConfig typed_config = 2;
+      }
+    }
+
+    // A linear set of match conditions that will be evaluated in order.
+    // The action of the first matcher that evaluates to true
+    // decides the action that will be performed. If no match
+    // criteria is matched, the no_match_action action will be performed.
+    repeated LeafMatcher matchers = 1;
+
+    // The action to perform if the predicate evaluates to true.
+    MatchAction action = 2;
+
+    MatchAction no_match_action = 3;
+  }
+
+  message SublinearMatcher {
+    message MultimapMatcher {
+      // The header to apply this match to.
+      // The key to apply this matcher for.
+      string key = 1;
+
+      // The namespace of this key. This is used to distinguish multiple maps
+      // that could be consulted during matching.
+      string namespace = 2;
+
+      // Mapping from exact value matches to the matching subtree to use when
+      // the match applies.
+      // If a header value matches both a prefix and an exact match, the exact
+      // match path will be taken.
+      map<string, MatchTree> exact_matches = 3;
```

Comment:
There is only exact in the PR, but earlier examples had prefix.

Reply:
I omitted for now since I didn't implement, let me add back in with not-implemented